app/models/ability.rb

```diff
@@ -93,6 +93,7 @@ def define_abilities
 
   def printer_abilities
     can :index, Printer if can_access_printers?
+    can [:create, :update], PrintersUser if can_access_printers?
   end
```

Comment:
```suggestion
  def printer_abilities
    if can_manage_printers?
      can :index, Printer
      can [:create, :update], PrintersUser
    elsif can_access_printers?
      can :index, Printer
      can [:create, :update], PrintersUser, { user_id: @user_id }
  end
```

I feel we need something like this. We can't have a non super-user modify someone else's printer

Reply:
Thoughts @steveyken

Reply:
Agreed. The boilerplate @patrixr set will work. Suggest using `can_update_my_printers?` so we know the permission is scoped to `user_id` and bring `can :index, Printer` out of the if block

```ruby
  def printer_abilities
    can :index, Printer
    if can_manage_printers?
      can [:create, :update], PrintersUser
    elsif can_update_my_printers?
       can [:create, :update], PrintersUser, { user_id: @user_id }
    end
  end
```